openwisp_monitoring/db/backends/influxdb/creation.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from .. import TIMESERIES_DB
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseCreation(object):
+    ORIGINAL_DB = TIMESERIES_DB['NAME']
+    TEST_DB = f'{ORIGINAL_DB}_test'
+
+    def create_database(self, database=TIMESERIES_DB['NAME']):
+        """ creates database if necessary """
+        db = self.get_db()
+        response = db.query('SHOW DATABASES')
+        items = list(response.get_points('databases'))
+        databases = [database['name'] for database in items]
+        # if database does not exists, create it
+        if database not in databases:
+            db.create_database(database)
+            logger.info(f'Created influxdb database {database}')
```

Comment:
What happens if we just create the database and it exists? Will it raise an exception? In this case I always prefer to “ask forgiveness not permission” and capture the exception. I feel it's also more pythonic

Reply:
Simply put, don't verify whether the database exists. In case influxdb raises an exception complaining db exists, capture and hide the exception, right? If so, I am not sure how does it help, does it speed up things significantly (unsure here). I will make the changes @PabloCastellano but it would be great if you can explain what's the clear advantage.

Reply:
[This is a common Python coding style](https://docs.python.org/3.4/glossary.html#:~:text=Easier%20to%20ask%20for%20forgiveness,many%20try%20and%20except%20statements.).

In terms of speed, well, it will save one query (SHOW DATABASES) when the database does not exist and will remain the same when the database exists. The code is also much shorter and hence less prone to errors

Reply:
from [influxdb-docs](https://docs.influxdata.com/influxdb/v1.8/query_language/manage-database/#create-database)
>A successful CREATE DATABASE query returns an empty result. If you attempt to create a database that already exists, InfluxDB does nothing and does not return an error.

So, I can safely create database I think without `look before you leap`, `ask for forgiveness not permission` using either coding styles :smile: